### DISCUSSION TITLE
How to create/update an MVAR table using FontTools

### DISCUSSION THREAD

**Discussion by djs-code** [fonttools/fonttools#3484] [Q&A] [answered]
Based on a provided variable font, how would one go about creating and/or updating an MVAR table that updates sTypoAscender and sTypoDescender depending on the value of FVAR axis `BLAH`?  The `BLAH` axis ranges from 0 to 1000, and the following variations would apply to sTypoAscender and sTypoDescender:

```
# <axis value>: (<sTypoAscender>, <sTypoDescender>)
variation_data = {
    0: (1424, -426),
    500: (1565, -468),
    1000: (1708, -511)
}
```

**Comment by behdad** [comment]
So you have an existing MVAR table that you want to modify? That's a bit tricky.

**Reply by djs-code** [reply]
Either-or.  I can start from not having an MVAR in the first place.

**Reply by behdad** [reply, marked as answer]
Here's the code in fontTools.varLib that builds the MVAR. Hopefully you can decipher it and achieve what you want.
